src/material/button/testing/shared.spec.ts

```diff
@@ -110,7 +123,12 @@ export function runHarnessTests(
     </button>
     <button id="raised" type="button" mat-raised-button>Raised button</button>
     <button id="stroked" type="button" mat-stroked-button>Stroked button</button>
-    <button id="icon" type="button" mat-icon-button>Icon button</button>
+    <button id="home-icon" type="button" mat-icon-button>
+      <mat-icon>home</mat-icon>
+    </button>
+    <button id="favorite-icon" type="button" mat-icon-button>
+      <mat-icon>favorite</mat-icon>
+    </button>
```

Comment:
Shouldn't one of these test cases remain as a `mat-icon-button` without a `mat-icon` element inside?
```suggestion
    <button id="home-icon" type="button" mat-icon-button>
      <mat-icon>home</mat-icon>
    </button>
    <button id="favorite-icon" type="button" mat-icon-button>favorite</button>
```

Reply:
I wanted to have two `mat-icon-buttons` with `mat-icons` inside so that we could make sure that the correct `mat-icon` was being returned.

Also, the `mat-icon` with text in it was being used for any tests. @mmalerba What are your thoughts here?

Reply:
I think the icon button with text in it was just a sloppy previously written test. I'm good with these changes